lib/resty/openssl/x509/csr.lua

```diff
@@ -162,6 +164,46 @@ function _M:to_PEM()
   return tostring(self, "PEM")
 end
 
+--- Get all csr extensions
+-- @tparam table self Instance of csr
+-- @treturn List of parsed extension objects
+function _M.get_extensions(self)
+    local extensions = C.X509_REQ_get_extensions(self.ctx)
+    local n = C.OPENSSL_sk_num(extensions)
```

Comment:
let's use the version independent wrapper from include/stack.lua instead, the symbol in 1.1.x and 1.0.x for stack is different. (see x509:get_ocsp_url as an example)

Reply:
i'm also wondering if it's a better idea to have `stack_of(X509_EXTENSION)` being wrapper as a lua table and use the stack macros here, like x509/chain.lua is been implemented. that can be a future improvement.